### ISSUE TITLE
Inputs inside overlay cannot keep focus

### ISSUE THREAD

**Issue by roginfarrer** [gpbl/react-day-picker#652]
Playing with the new `keepFocus` prop, I'm finding a bug that's exposed when using inputs inside the overlay. Basically, if the input is controlled, and the value is changed, the entire overlay re-renders, so the input loses focus. It doesn't seem like a case where the overlay should re-render?

Would be great to come up with a solution for this (possibly with user implementation) for folks like me who are trying to build a time-picker inside the overlay.

I'll try to look into this when I have time.

[CodeSandbox with demo](https://codesandbox.io/s/62r9vvk653)

To replicate: open the overlay, focus on the input, and try to edit the number inside it. The input will lose focus immediately.

![Gif](https://cl.ly/pxKp/Screen%20Recording%202018-03-05%20at%2005.09%20PM.gif)

**Comment by gpbl** [comment]
🤔 A solution I see here is to focus again the overlay's input. Maybe a `shouldComponentUpdate` in  `CustomOverlay` could help. Also, I don't think it is a good idea to add a time picker inside the overlay 😄

**Comment by roginfarrer** [comment]
I discovered the issue comes from giving `overlayComponent` an inline function instead of a class. So it will re-render on every state change:

```jsx
// no re-render on state change
overlayComponent: CustomOverlay

// Given props on render, will re-run function on state change
overlayComponent={props => (
  <CustomOverlay
    {...props}
    onChange={this.handleInputChange}
    inputValue={this.state.value}
    refCallback={this.assignRef}
  />
)}
```

The only way I can think to avoid needing to provide props on run time is another prop like `overlayProps` that's passed in the `Overlay` component in `DayPickerInput.js`.

@dialnco's idea is a good one, though. I'll run his solution by my team. (Thanks!)

**Comment by gpbl** [comment]
With the recent fix landed in `v7.1.8` now this is possible again. You need however to use `keepFocus={false}`. 

See this example showing an input field in the `overlayComponent`: http://react-day-picker.js.org/examples/input-custom-overlay

**Comment by dheerajsinghnagdali** [comment]
I still don't understand how I can achieve this.
